packages/permissions/src/utils/has-permissions.ts

```diff
@@ -141,3 +172,71 @@ export const getInvalidPermissions = (
     isNil(actualPermissions[toCanCase(demandedPermission)])
   );
 };
+
+const hasDemandedDataFenceForStore = (
+  resourceToApplyDataFence: TResourceToApplyDataFence,
```

Comment:
Here we could then do `authorizationContext` which is `TResourceToApplyDataFence`. That might even nicely work then with type guards in teh future.